rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -1095,6 +1105,57 @@ static T ppu_load_acquire_reservation(ppu_thread& ppu, u32 addr)
 	ppu.raddr = addr;
 	const u64 mask_res = g_use_rtm ? (-128 | vm::dma_lockb) : -1;
 
+	if (const s32 max = g_cfg.core.ppu_128_reservations_loop_max_length)
+	{
+		// If we use it in HLE it means we want the accurate version
+		ppu.use_full_rdata = max < 0 || ppu.current_function || [&]()
```

Comment:
current_function may remain set within callbacks... But I don't see why it should ever be checked. Why would we need to invoke it from HLE code?

Reply:
In places where we to implement LLE-like accuracy in HLE code and load<->cmpxchg loop is just not enough, its "alright" if it may also being hit rarely in some callbacks code regardless of the setting's value.
Although I think ppu_thread::fast_call does set current_function to nullptr (used in callbacks) ? and then restore the old value when it returns.

Reply:
Live example of a place where it matters: and rpcs3 implemented it wrongly using load-cmpxchg loop https://github.com/RPCS3/rpcs3/blob/master/rpcs3/Emu/Cell/Modules/cellSync.cpp#L1551
Notice how push3's value does not change yet the atomic op executes with no real effects, I assume this was an equivalent of LLE code where stwcx is used but its used to ensure all the previous loads were atomic and no atomic updates occured inbetween them.
Infact this is also true about the rest of the functions here as well.

Reply:
I just don't see actual scenario of using ldarx/stdcx in actual HLE code. It's clunky and there are better tools for it.

Reply:
On the implementation of HLE cellSpursAddUrgentCommand I used it to properly synchronize with SPU code which supposedly won't be HLEd as well so its the only proper option available.
I mean, HLEing all these functions may not be a thing for a while but its a feature which needs to exist if someone wants need it, in which case the code won't try analyze the memory pointed by PPU CIA and by default use the accurate version.